.github/workflows/pytest.yml

```diff
@@ -0,0 +1,214 @@
+            os: ubuntu
+            conda: scipy numba
+            pip: ""
+            test: ""
+          - python-version: 3.7
+            name: full
+            os: ubuntu
+            conda: scipy numba matplotlib
+            pip: pytest-mpl scooby
+            test: "--mpl"
+          - python-version: 3.8
+            name: plain
+            os: ubuntu
+            conda: scipy numba
+            pip: ""
+            test: ""
+          - python-version: 3.8
+            name: full
+            os: ubuntu
+            conda: scipy numba matplotlib
+            pip: pytest-mpl scooby
+            test: "--mpl"
+
+    env:
+      # Used to for coveralls flag
+      OS: ${{ matrix.case.os }}
+      PYTHON: ${{ matrix.case.python-version }}
+
+    steps:
+
+      # Cancel any previous run of the test job; [pin v0.6.0]
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@ce177499ccf9fd2aded3b0426c97e5434c2e8a73
+        with:
+          access_token: ${{ github.token }}
+
+      # Checks-out your repository under $GITHUB_WORKSPACE
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          # Need to fetch more than the last commit so that setuptools_scm can
+          # create the correct version string. If the number of commits since
+          # the last release is greater than this, the version still be wrong.
+          # Increase if necessary.
+          fetch-depth: 100
+          # The GitHub token is preserved by default but this job doesn't need
+          # to be able to push to GitHub.
+          persist-credentials: false
+
+      # Need the tags so that setuptools_scm can form a valid version number
+      - name: Fetch git tags
+        run: git fetch origin 'refs/tags/*:refs/tags/*'
+
+      - name: Setup miniconda
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+          auto-update-conda: true
+          miniconda-version: "latest"
+          mamba-version: "*"
+          python-version: ${{ matrix.case.python-version }}
+          activate-environment: tenv
+          auto-activate-base: false
+          channels: conda-forge,defaults
+          channel-priority: strict
+
+      - name: Install dependencies
+        shell: bash -l {0}
+        run: |
+          conda config --set always_yes yes --set changeps1 no
+          conda config --show-sources
+          conda config --show
+          mamba install ${{ matrix.case.conda }} pytest pytest-cov pip
+          pip install setuptools_scm coveralls pytest-flake8 ${{ matrix.case.pip }}
+          conda info -a
+          conda list
+
+      - name: Test with pytest
+        shell: bash -l {0}
+        run: |
+          python setup.py install
+          cp ./tests/matplotlibrc ./.
+          pytest --cov=empymod --flake8 ${{ matrix.case.test }}
+
+      - name: Coveralls
+        # [pin v20201129]
+        uses: AndreMiras/coveralls-python-action@f5fd5c309b39d01599fb92c72d4f7409ea78aec9
```

Comment:
Why not using the tag directly?
```yaml
        uses: AndreMiras/coveralls-python-action@v20201129
```

Reply:
For security reasons. If your GitHub is compromised, someone could delete your release `v20201129`, create another release with the tag `v20201129`, and put malicious code in it. At least that is how I understood this:

https://github.com/pypa/gh-action-pypi-publish/issues/27

I got a lot of my Action-yaml-ideas from https://github.com/fatiando/boule/blob/master/.github/workflows/pypi.yml - so @leouieda looked probably deeper into this issue and will correct me ;-)

Reply:
I guess the important / more extensive link is in the comment of the GitHub issue I posted: https://julienrenaux.fr/2019/12/20/github-actions-security-risk/

Reply:
Yeah, I’m a bit paranoid 🙂 I don’t much like the idea of giving a token to any action author out there. Not that I think you’re malicious @andremiras. But there are a lot of actions and I find it worrisome that GitHub has no control over this. It’s inevitable that someone will make an action at some point that steals tokens.

Reply:
I understand your concern, isn't computer security all about being paranoid?
It's also about finding the right balance between security and productivity while keeping in mind the risk (impact/likelihood).
I for now prefer to keep it simple and readable with the tag, but I'm happy this potential flaw got brought to my attention :smiley:

Reply:
You're absolutely right about needing a balance. For me, I've decided to use the commit hash whenever giving a token to an Action. I'm also not too worried about the official GitHub actions.